be/src/runtime/tablets_channel.cpp

```diff
@@ -187,6 +188,17 @@ void TabletsChannel::add_chunk(brpc::Controller* cntl, const PTabletWriterAddChu
                 delta_writer->commit(cb);
             }
         }
+
+        // persist txn.
+        auto tablet_ids = request.tablet_ids();
```

Comment:
delta_writer->commit() is async

Reply:
Updated. The persisting of txn would be done after the `delta_writer-commit()` is done.